iterators task (1)/student/include/impl.h

```diff
@@ -5,11 +5,51 @@ template <class Container>
 struct accumulator
 {
     /** @todo Iterator traits*/
+    using iterator_category = typename std::output_iterator_tag;
+    using value_type        = Container;
+    using difference_type   = std::ptrdiff_t;
+    using pointer           = value_type*;
+    using reference         = value_type&;
+
     /** @todo Constructor with container*/
+    accumulator(Container& container) : mContainer(&container) {}
     /** @todo Copy assignment operator*/
+    accumulator<Container>(const accumulator<Container>&) = default;
+    accumulator<Container>& operator=(const accumulator<Container>&) = default;
     /** @todo Move assignment operator*/
+    accumulator<Container>(accumulator<Container>&& val) = default;
+    accumulator<Container>& operator=(accumulator<Container>&& val) = default;
 
     /** @todo Iterator operations */
+    accumulator<Container>& operator++()
+    {
+        return *this;
+    }
+    accumulator<Container>& operator*()
+    {
+        return *this;
+    }
+
+    ///Mixin interface for an iterator & reference type is not
+    ///a canonical way to implement iterators.
+    ///Nevertheless it is done so for the sake of simplicity.
```

Comment:
Please provide more info, do you have any suggestion for this task?

Reply:
it looks odd that operator* returns pointer to itself, but not to the reference:
it could be either T& or some adapter(ie map,vec<bool>)

Reply:
in any case it would put assignment  operators to a more appropriate context.

Reply:
Still can't understand, could you please highlight the problem?
operator* should return dereferenced pointer of _this_, so return type is reference

Reply:
even the current iface fulfills requirements to iterator. it is inconsistent. ie. operator* is redundant, as we can assign c2 directly to iterator. basically it breaks single responsibility principle.